tests/test_evpn_l3_vxlan.py

```diff
@@ -0,0 +1,1726 @@
+        print "\tPass"
+
+        print "\tTesting VLAN 100 extension to 8.8.8.8 and 7.7.7.7"
+        vxlan_obj.check_vlan_extension(dvs, '100', '8.8.8.8')
+        vxlan_obj.check_vlan_extension(dvs, '100', '7.7.7.7')
+        print "\tPass"
+
+        print "\tTesting VLAN 100 interface creation"
+        create_vlan_interface(dvs, "Vlan100", "Ethernet24", "Vrf-RED", "100.100.3.1/24")
+        vxlan_obj.check_router_interface(dvs, 'Vrf-RED', vxlan_obj.vlan_id_map['100'], 2)
+        print "\tPass"
+
+        print "\tTest VRF IPv4 Route with Tunnel Nexthop 7.7.7.7 Add"
+        vxlan_obj.fetch_exist_entries(dvs)
+        create_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED', '7.7.7.7', "Vlan100", "00:11:11:11:11:11", '1000')
+        vxlan_obj.check_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED', '7.7.7.7', tunnel_name, "00:11:11:11:11:11", '1000')
+        print "\tPass"
+
+        print "\tTest VRF IPv4 Route with Tunnel Nexthop 7.7.7.7 Delete"
+        delete_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED')
+        vxlan_obj.check_del_tunnel_nexthop(dvs, 'Vrf-RED', '7.7.7.7', tunnel_name, "00:11:11:11:11:11", '1000')
+        vxlan_obj.check_del_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED')
+        print "\tPass"
+
+        print "\n\nTesting IPv4 Route and Overlay Nexthop Update"
+        print "\tTest VRF IPv4 Route with Tunnel Nexthop 7.7.7.7 Add"
+        vxlan_obj.fetch_exist_entries(dvs)
+        create_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED', '7.7.7.7', "Vlan100", "00:11:11:11:11:11", '1000')
+        vxlan_obj.check_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED', '7.7.7.7', tunnel_name, "00:11:11:11:11:11", '1000')
+        print "\tPass"
+
+        print "\tTest Tunnel Nexthop change from 7.7.7.7 to 8.8.8.8"
+        create_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED', '8.8.8.8', "Vlan100", "00:22:22:22:22:22", '1000')
+        vxlan_obj.check_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED', '8.8.8.8', tunnel_name, "00:22:22:22:22:22", '1000', 1)
+        print "\tPass"
+
+        print "\tTest Previous Tunnel Nexthop 7.7.7.7 is removed"
+        vxlan_obj.check_del_tunnel_nexthop(dvs, 'Vrf-RED', '7.7.7.7', tunnel_name, "00:11:11:11:11:11", '1000')
+        print "\tPass"
+
+        print "\tTest VRF IPv4 Route and Tunnel Nexthop 8.8.8.8 Delete"
+        vxlan_obj.fetch_exist_entries(dvs)
+        delete_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED')
+        vxlan_obj.check_del_tunnel_nexthop(dvs, 'Vrf-RED', '8.8.8.8', tunnel_name, "00:22:22:22:22:22", '1000')
+        vxlan_obj.check_del_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED')
+        print "\tPass"
+
+        print "\n\nTest VRF IPv4 Route with ECMP Tunnel Nexthop Add and Delete"
+        vxlan_obj.fetch_exist_entries(dvs)
+
+        ecmp_nexthop_attr = [
+            ("nexthop", "7.7.7.7,8.8.8.8"),
+            ("ifname", "Vlan100,Vlan100"),
+            ("vni_label", "1000,1000"),
+            ("router_mac", "00:11:11:11:11:11,00:22:22:22:22:22"),
```

Comment:
How do we verify the local router_mac? ie., local routes getting advertised with local router_mac? Can we test with local static routes?

Reply:
I did not get the query. Please elaborate.

This is EVPN route over tunnel test case. Static route with tunnel nexthop is currently not supported.